kubectl-fdb/cmd/k8s_client.go

```diff
@@ -294,6 +295,33 @@ func getProcessGroupIDsFromPodName(cluster *fdbv1beta2.FoundationDBCluster, podN
 	return processGroupIDs, nil
 }
 
+// fetchProcessGroupsCrossCluster fetches the list of process groups matching the given podNames and returns the
+// processGroup objects mapped by clusterName matching the given clusterLabel.
+func fetchProcessGroupsCrossCluster(kubeClient client.Client, namespace string, clusterLabel string, podNames ...string) (map[*fdbv1beta2.FoundationDBCluster][]fdbv1beta2.ProcessGroupID, error) {
+	var pod corev1.Pod
+	pgsByCluster := map[*fdbv1beta2.FoundationDBCluster][]fdbv1beta2.ProcessGroupID{}
+
+	for _, podName := range podNames {
+		err := kubeClient.Get(context.Background(), client.ObjectKey{Name: podName, Namespace: namespace}, &pod)
+		if err != nil {
+			if k8serrors.IsNotFound(err) {
+				return nil, fmt.Errorf("could not get pod: %s/%s", namespace, podName)
+			}
+			return nil, err
+		}
+		clusterName := pod.Labels[clusterLabel]
```

Comment:
Good catch!

Reply:
Leaning towards aborting entirely rather than skipping the one pod (and allowing the user to remove the pod from the list if there is a problem with the pod and not the label), let me know if you disagree